app/models/valid_datacite.rb

```diff
@@ -0,0 +1,141 @@
+      # @param publication_year [Integer] year when the resource is made publicly available.
+      # @param subjects [Array<Subject>] subjects, keywords, classification codes, or key phrases describing the resource.
+      # @param funding_references [Array<FundingReference>] information about financial support (funding) for the resource being registered.
+      # @param contributors [Array<Contributor>] institutions or persons responsible for collecting, creating, or otherwise contributing to the developement of the dataset.
+      # @param dates [Array<Date>] different dates relevant to the work.
+      # @param language [String, nil] Primary language of the resource: an IETF BCP 47, ISO 639-1 language code.
+      # @param resource_type [ResourceType, nil] the type of the resource
+      # @param alternate_identifiers [Array<AlternateIdentifier>] an identifier or identifiers other than the primary {Identifier} applied to the resource being registered.
+      # @param related_identifiers [Array<RelatedIdentifier>] identifiers of related resources.
+      # @param sizes [Array<String>] unstructured size information about the resource.
+      # @param formats [Array<String>] technical format of the resource, e.g. file extension or MIME type.
+      # @param version [String] version number of the resource.
+      # @param rights_list [Array<Rights>] rights information for this resource.
+      # @param descriptions [Array<Description>] all additional information that does not fit in any of the other categories.
+      # @param geo_locations [Array<GeoLocations>] spatial region or named place where the data was gathered or about which the data is focused.
+      @mapping = Datacite::Mapping::Resource.new(
+        identifier: datacite_identifier(@metadata_from_form["identifier"]),
+        creators: datacite_creators,
+        titles: datacite_titles,
+        publisher: datacite_publisher(@metadata_from_form["publisher"]),
+        publication_year: @metadata_from_form["publication_year"],
+        # subjects: [],
+        # contributors: [],
+        # dates: [],
+        # language: nil,
+        # funding_references: [],
+        resource_type: datacite_resource_type(@metadata_from_form["resource_type"])
+        # alternate_identifiers: [],
+        # related_identifiers: [],
+        # sizes: [],
+        # formats: [],
+        # version: nil,
+        # rights_list: [],
+        # descriptions: [],
+        # geo_locations: []
+      )
+      @mapping
+    end
+
+    def to_xml
+      datacite_mapping.write_xml
+    end
+
+    ##
+    # For each creator, make a Datacite::Mapping::Creator object
+    # TODO: This class has no field for author order. We need a place to record that.
+    def datacite_creators
+      creator_array = []
+      @metadata_from_form["creators"].each do |creator|
```

Comment:
Could this just be a map?  Then you would not need the separate array and it would be clearer that it is a one to one relationship.
```suggestion
      @metadata_from_form["creators"].map do |creator|
```